libs/dh/esett/feature-outgoing-messages/src/lib/drawer/dh-outgoing-message-drawer.component.ts

```diff
@@ -183,4 +190,46 @@ export class DhOutgoingMessageDrawerComponent {
       },
     });
   }
+
+  openResolveModal() {
+    this.modalService.open({
+      component: DhResolveModalComponent,
+      data: { resolve: this.resolve },
+    });
+  }
+
+  resolve = async (comment: string): Promise<boolean> => {
+    if (!this.outgoingMessage?.documentId) {
+      return false;
+    }
+
+    const result = await this.resolveMutation.mutate({
+      variables: { input: { documentId: this.outgoingMessage.documentId, comment } },
+    });
+
+    if (result.error) {
+      this.toastService.open({
+        type: 'danger',
+        message: translate('eSett.outgoingMessages.drawer.resolveModal.resolvedError'),
+      });
+      return false;
+    }
+
+    this.toastService.open({
+      type: 'success',
+      message: translate('eSett.outgoingMessages.drawer.resolveModal.resolvedSuccess'),
+    });
+
+    this.loadOutgoingMessage(this.outgoingMessage.documentId);
```

Comment:
Will this reload the status in list too?

Reply:
You can use refetchQueries on the mutations so you dont need to refetch

https://github.com/Energinet-DataHub/greenforce-frontend/blob/1c0e568d4e8d77735583a55e59147186da4fd91f/libs/dh/profile/feature-profile-modal/src/lib/dh-profile-modal/dh-profile-modal.component.ts#L136